### PR TITLE
TECH-1996: Create ARCH version of spicerack-styleguide rubocop.yml

### PR DESCRIPTION
**Pull request by tbconroy**
https://freshly.atlassian.net/browse/TECH-1996

We want to subclass the rubocop.yml in the spicerack-styleguide to override cops in the ARCH/BL repos that otherwise wouldn't be feasible in the greater Freshly engineering ecosystem.

The usage for rubocop.yml, now essentially the base config, would still be the same. However the ARCH repos, and anyone else who wants to suffer under our strong code style opinions, would be able to use rubocop.arch.yml for slightly tweaked rules. Is there a better way to do this?

This PR/branch is almost identical to #342 but breaks up the commits/uses `git mv` to preserve git history

### CODE REVIEW

File: spicerack-styleguide/rubocop.arch.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+inherit_from: rubocop.base.yml
+
+ # Cop overrides specific to the ARCH pod:
+ Metrics/LineLength:
```
Comment (vinodlala) [inline spicerack-styleguide/rubocop.arch.yml]: Remove leading space here.

So while I have your branch in my spicerack repo:

```
vinodlala@NY-MACLT-124 payment_service (develop)*$ bundle exec rubocop -c ../spicerack/spicerack-styleguide/rubocop.arch.yml
(/Users/vinodlala/Freshly_old/spicerack/spicerack-styleguide/rubocop.arch.yml): did not find expected key while parsing a block mapping at line 1 column 1
```

Comment (vinodlala) [reply spicerack-styleguide/rubocop.arch.yml]: nvm, I made a commit to fix it.

Comment (tbconroy) [reply spicerack-styleguide/rubocop.arch.yml]: Thanks!

Comment (tbconroy) [reply spicerack-styleguide/rubocop.arch.yml]: Also, doh! 😬

**Pull request merged**